server/src/main/java/org/elasticsearch/repositories/RepositoryShardId.java

```diff
@@ -13,64 +13,23 @@
 import org.elasticsearch.common.io.stream.Writeable;
 
 import java.io.IOException;
-import java.util.Objects;
 
 /**
  * Represents a shard snapshot in a repository.
  */
-public final class RepositoryShardId implements Writeable {
+public record RepositoryShardId(IndexId index, int shardId) implements Writeable {
 
-    private final IndexId index;
-
-    private final int shard;
-
-    public RepositoryShardId(IndexId index, int shard) {
-        assert index != null;
-        this.index = Objects.requireNonNull(index);
```

Comment:
It looks like `requireNonNull` is lost. Should this have:
```
public RepositoryShardId {
  Objects.requireNonNull(index);
}
```
?

Reply:
I don't think it matters much or at all. I looked over this quickly and at the very latest this would fail during `writeTo` (but also earlier) if we ever ran into a `null` here. Seemed like it wasn't worth the clutter to me so I left it out.